consensus/istanbul/validator/multi_staking_test.go

```diff
@@ -0,0 +1,317 @@
+// Copyright 2019 The klaytn Authors
+// This file is part of the klaytn library.
+//
+// The klaytn library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The klaytn library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the klaytn library. If not, see <http://www.gnu.org/licenses/>.
+
+/*
+Multiple Staking Contracts
+
+Validators can deploy multiple staking contracts.
+If a validator wants to deploy additional staking contracts, those staking contracts should have same rewardAddress.
+StakingAmounts of staking contracts with a same rewardAddress will be added and it is reflected to a probability of becoming a block proposer.
+
+Testing
+
+StakingInfos are data from addressBook.
+A StakingInfo has lists of addresses and stakingAmount.
+They are matched by an index. Values of the lists with a same index are from a same staking contract.
+
+All addresses used in tests are made by 3 digits number.
+NodeAddress : begin with 1
+rewardAddress : begin with 2
+NodeAddress of additional staking contract : begin with 9
+*/
+package validator
+
+import (
+	"github.com/klaytn/klaytn/common"
+	"github.com/klaytn/klaytn/consensus/istanbul"
+	"github.com/klaytn/klaytn/reward"
+	"github.com/stretchr/testify/assert"
+	"math/big"
+	"testing"
+)
+
+func newTestWeightedCouncil(nodeAddrs []common.Address) *weightedCouncil {
+	return NewWeightedCouncil(nodeAddrs, nil, make([]uint64, len(nodeAddrs)), nil, istanbul.WeightedRandom, 0, 0, 0, nil)
+}
+
+// check if validators and stakingAmount from stakingInfo is matched well.
+// stakingAmounts of additional staking contracts will be added to stakingAmounts of validators which have the same reward address.
+// input
+//  - validator and stakingInfo is matched by a nodeAddress.
+// output
+//  - weightedValidators are sorted by nodeAddress
+//  - stakingAmounts should be same as expectedStakingAmounts
+func TestWeightedCouncil_getStakingAmountsOfValidators(t *testing.T) {
+	testCases := []struct {
+		validators             []common.Address
+		stakingInfo            *reward.StakingInfo
+		expectedStakingAmounts []float64
+	}{
+		{
+			[]common.Address{common.BigToAddress(big.NewInt(101)), common.BigToAddress(big.NewInt(102)), common.BigToAddress(big.NewInt(103))},
```

Comment:
We can use `StringToAddress(s string)` instead of BigToAddress?

Reply:
or common.Address{0}, common.Address{1}??

Reply:
reflected. changed to StringToAddress(s string)